ruby/lib/ci/queue/redis/build_record.rb

```diff
@@ -59,20 +59,10 @@ def record_warning(type, attributes)
         Test = Struct.new(:id) # Hack
 
         def record_error(id, payload, stats: nil)
-          # FIXME: the ack and hset should be atomic
-          # otherwise the reporter may see the empty queue before the error
-          # is appended and wrongly think it's a success.
-          if @queue.acknowledge(Test.new(id))
-            redis.pipelined do |pipeline|
-              pipeline.hset(
-                key('error-reports'),
-                id.dup.force_encoding(Encoding::BINARY),
```

Comment:
@casperisfine do you remember why we sometimes encode the test id (`id.dup.force_encoding(Encoding::BINARY)`) and other times we don't?

Reply:
To workaround a bug in redis that has since then been fixed: https://github.com/Shopify/ci-queue/issues/84